src/languages/dos.js

```diff
@@ -12,9 +12,15 @@ export default function(hljs) {
     /^\s*@?rem\b/, /$/,
     { relevance: 10 }
   );
+
+  // for matching comments starting with ::
+  const COMMENT_2 = hljs.COMMENT(
+    /^::/, /$/,
+    { relevance: 10 }
+  );
   const LABEL = {
     className: 'symbol',
-    begin: '^\\s*[A-Za-z._?][A-Za-z0-9_$#@~.?]*(:|\\s+label)',
+    begin: '^:',
```

Comment:
Wouldn't this be something more like:

```
/^:[A-Za-z._?][A-Za-z0-9_$#@~.?]*/
```

Or whatever value characters for a label are?

Reply:
I'll have to take a closer look to see how this is currently working.  This grammar pattern is pretty old... I think we could have a much simpler rule here, or perhaps a multi-match if we truly want the `:` to be different than the name of the label.